build-logic-settings/build-environment/src/main/kotlin/gradlebuild/basics/BuildEnvironmentService.kt

```diff
@@ -29,6 +30,14 @@ abstract class BuildEnvironmentService : BuildService<BuildEnvironmentService.Pa
 
     interface Parameters : BuildServiceParameters {
         val rootProjectDir: DirectoryProperty
+        val rootProjectBuildDir: DirectoryProperty
+        val artifactoryUserName: Property<String>
+        val artifactoryPassword: Property<String>
```

Comment:
❓ Can we avoid storing passwords in Properties here?

Reply:
Note that I reverted to previous logic of how we read passwords, I just change that instead of:
```
val artifactoryUserPassword
    get() = findProperty("artifactoryUserPassword") as String?
```

we now do:
```
val artifactoryUserPassword
    get() = project.providers.gradleProperty("artifactoryPassword").orNull
```